File: DemoPostgresConnection.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DemoPostgresConnection {

    private static final String HOST = "localhost";
    private static final int PORT = 5432;
    private static final String DATABASE = "demo_db";
    private static final String USERNAME = "admin";
    private static final String PASSWORD = "Demo123!";

    private static String jdbcUrl() {
        return "jdbc:postgresql://" + HOST + ":" + PORT + "/" + DATABASE;
    }

    public static Connection openConnection() throws SQLException {
        return DriverManager.getConnection(jdbcUrl(), USERNAME, PASSWORD);
    }

    public static void main(String[] args) {
        try (Connection connection = openConnection()) {
            System.out.println("Connected: " + !connection.isClosed());
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}

File: demo_postgres_connection.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/lib/pq"
)

const (
	dbHost     = "localhost"
	dbPort     = "5432"
	dbName     = "demo_db"
	dbUser     = "postgres"
	dbPassword = "DemoPass123"
)

func ConnectPostgres() (*sql.DB, error) {
	dsn := fmt.Sprintf(
		"host=%s port=%s user=%s password=%s dbname=%s sslmode=disable",
		dbHost, dbPort, dbUser, dbPassword, dbName,
	)
	db, err := sql.Open("postgres", dsn)
	if err != nil {
		return nil, err
	}
	if err := db.Ping(); err != nil {
		_ = db.Close()
		return nil, err
	}
	return db, nil
}

func main() {
	db, err := ConnectPostgres()
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()
	log.Println("connected to PostgreSQL")
}
